src/alpaka/test/alpaka/AtomicPairCounter_t.cc

```diff
@@ -72,66 +64,60 @@ struct verify {
 
 int main() {
   const DevHost host(alpaka::pltf::getDevByIdx<PltfHost>(0u));
-  const DevAcc device(alpaka::pltf::getDevByIdx<PltfAcc>(0u));
+  const DevAcc1 device(alpaka::pltf::getDevByIdx<PltfAcc1>(0u));
   Queue queue(device);
 
   constexpr uint32_t C = 1;
-  const Vec sizeC(C);
-  auto c_dbuf = alpaka::mem::buf::alloc<cms::Alpaka::AtomicPairCounter, Idx>(device, sizeC);
+  const Vec1 sizeC(C);
+  auto c_dbuf = alpaka::mem::buf::alloc<cms::alpakatools::AtomicPairCounter, Idx>(device, sizeC);
   alpaka::mem::view::set(queue, c_dbuf, 0, sizeC);
 
-  std::cout << "size " << C * sizeof(cms::Alpaka::AtomicPairCounter) << std::endl;
+  std::cout << "size " << C * sizeof(cms::alpakatools::AtomicPairCounter) << std::endl;
 
   constexpr uint32_t N = 20000;
   constexpr uint32_t M = N * 6;
-  const Vec sizeN(N);
-  const Vec sizeM(M);
+  const Vec1 sizeN(N);
+  const Vec1 sizeM(M);
   auto n_dbuf = alpaka::mem::buf::alloc<uint32_t, Idx>(device, sizeN);
   auto m_dbuf = alpaka::mem::buf::alloc<uint32_t, Idx>(device, sizeM);
 
   constexpr uint32_t NUM_VALUES = 10000;
 
-  // Prepare 1D workDiv
-  Vec elementsPerThread(Vec::all(1));
-  Vec threadsPerBlock(Vec::all(512));
-  const Vec blocksPerGrid(Vec::all(2000));
-#if defined ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED || ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLED || \
-    ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLED || ALPAKA_ACC_CPU_BT_OMP4_ENABLED
-  // on the GPU, run with 32 threads in parallel per block, each looking at a single element
-  // on the CPU, run serially with a single thread per block, over 32 elements
-  std::swap(threadsPerBlock, elementsPerThread);
-#endif
-  const WorkDiv workDiv(blocksPerGrid, threadsPerBlock, elementsPerThread);
-
+  // Update
+  const Vec1 &blocksPerGrid(Vec1(2000u));
```

Comment:
Yes for a diagonal read, I find it easier to see that the constructor is called in `const Vec1& blocksPerGrid(Vec1(2000u));` .
`const Vec1 blocksPerGrid(2000u);` can look like a function call from far away (even if there would be a `. `, a '->', or a `:: `in front of blocksPerGrid in that case).
But I agree this is a bit overkill, can change it if you like.

Reply:
Very likely compiler will optimize the cost away, but literally it reads now "a `const` reference to a temporary", and in principle in a (bad) implementation all accesses could go through an indirection. Anyway, object initialization is a confusing topic in C++ in general, and this is not very important for the prototyping.

Reply:
> `const Vec1& blocksPerGrid(Vec1(2000u));`
> Very likely compiler will optimize the cost away, but literally it reads now "a const reference to a temporary", and in principle in a (bad) implementation all accesses could go through an indirection. 

Are you sure? This is not a reference to a temporary, but a const-reference to a temporary.
When the temporary is bound to a **const** reference, it has its lifetime extended to the lifetime of the reference. So I think there is actually never a call to move/copy constructor anyway: the **destructor of the temporary is just called later** (at the reference end of scope). This should be guaranteed by the standard, and would not be compiler-dependent.
So I would think that `const Vec1& blocksPerGrid(Vec1(2000u));` is actually pretty similar to `const Vec1 blocksPerGrid 2000u);`.
But I agree the extra 'Vec1' is overkill (just find it nicer to read), so not worth the hassle. I have used the simple version in the HistoContainer PR.

Instead, in `const auto blocksPerGrid = Vec1(2000u);` : 
Move/copy constructor can be called, even if most likely more/copy are elided by the compiler anyway. Still, that makes it compiler-dependent.

Reply:
> So I would think that `const Vec1& blocksPerGrid(Vec1(2000u));` is actually pretty similar to `const Vec1 blocksPerGrid (2000u);`.

Yes, and I think having the reference there is confusing, because just using the object itself would be at least as good, and it would be simpler to read and write.

> Instead, in `const auto blocksPerGrid = Vec1(2000u);` :
> Move/copy constructor can be called, even if most likely more/copy are elided by the compiler anyway. Still, that makes it compiler-dependent.

C++17 requires copy elision, and cppreference says the following for [copy initialization](https://en.cppreference.com/w/cpp/language/copy_initialization)
>  if T is a class type and the initializer is a prvalue expression whose cv-unqualified type is the same class as T, the initializer expression itself, rather than a temporary materialized from it, is used to initialize the destination object: see [copy elision](https://en.cppreference.com/w/cpp/language/copy_elision)

Reply:
Oki

Reply:
Ha yes true I forgot copy elision is now guaranteed by the standard anyway